SwrveSDK/SDK/Track/Swrve.m

```diff
@@ -2201,6 +2235,8 @@ - (void) eventsSentCallback:(enum HttpStatus)status withData:(NSData*)data andCo
                 DebugLog(@"Error sending event data to Swrve (%@) Adding data back onto unsent message buffer", [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding]);
                 [[swrve eventBuffer] addObjectsFromArray:[client_info buffer]];
                 [swrve setEventBufferBytes:[swrve eventBufferBytes] + [client_info bufferLength]];
+                [swrve saveEventsToDisk];
+                [self setEventsWereSent:NO];
```

Comment:
Don't think this should be set here. There could be another thread that sets it to YES and thus the refreshcampaigns tick might not happen because of this NO when its suppose to.